compiler/luci/service/src/Nodes/CircleFullyConnected.test.cpp

```diff
@@ -59,3 +61,111 @@ TEST(CloneNodeTest, clone_FullyConnected_wf_NEG)
   auto cloned = luci::clone_node(node_fc, gc.get());
   ASSERT_EQ(nullptr, cloned);
 }
+
+TEST(ShapeRuleTest, fully_connected_dynamic_shape_keep_dims)
+{
+  luci::CircleInput input;
+  luci::CircleConst weights;
+  luci::CircleConst bias;
+  luci::CircleFullyConnected fully_connected;
+
+  input.shape({1, 10, 20});
+  input.shape_status(luci::ShapeStatus::VALID);
+  input.dim(1).unset();
+
+  weights.shape({15, 20});
+  weights.shape_status(luci::ShapeStatus::VALID);
+
+  bias.shape_status(luci::ShapeStatus::VALID);
+
+  fully_connected.input(&input);
+  fully_connected.weights(&weights);
+  fully_connected.bias(&bias);
+  fully_connected.keep_num_dims(true);
+
+  loco::TensorShape shape;
+  luci::sinf::Rule shape_inf_rule;
+
+  ASSERT_TRUE(shape_inf_rule.infer(&fully_connected, shape));
+  ASSERT_EQ(3, shape.rank());
+  ASSERT_TRUE(shape.dim(0).known());
+  ASSERT_FALSE(shape.dim(1).known());
+  ASSERT_TRUE(shape.dim(2).known());
+
+  ASSERT_EQ(1, shape.dim(0).value());
+  ASSERT_EQ(15, shape.dim(2).value());
```

Comment:
Maybe `dim(1)` missed? :)
```suggestion
  ASSERT_EQ(10, shape.dim(1).value());
  ASSERT_EQ(15, shape.dim(2).value());
```